Custom createParticles not compatible LPS

Note error:
[ 58%] Building CXX object examples/mechanics/CMakeFiles/FragmentingCylinder.dir/fragmenting_cylinder.cpp.o
CabanaPD/examples/mechanics/fragmenting_cylinder.cpp:108:16: error: no matching member function for call to 'createParticles'
    particles->createParticles( exec_space(), init_op );
    ~~~~~~~~~~~^~~~~~~~~~~~~~~
CabanaPD/src/CabanaPD_Particles.hpp:563:10: note: candidate function template not viable: requires single argument 'exec_space', but 2 arguments were provided
    void createParticles( const ExecSpace& exec_space )
         ^
1 error generated.
make[2]: *** [examples/mechanics/CMakeFiles/FragmentingCylinder.dir/fragmenting_cylinder.cpp.o] Error 1
make[1]: *** [examples/mechanics/CMakeFiles/FragmentingCylinder.dir/all] Error 2
make: *** [all] Error 2

--- a/src/CabanaPD_Particles.hpp
+++ b/src/CabanaPD_Particles.hpp
@@ -565,8 +565,8 @@ class Particles<MemorySpace, LPS, TemperatureIndependent, Dimension>
         // Forward arguments to standard or custom particle creation.
         base_type::createParticles( std::forward<Args>( args )... );
         _init_timer.start();
-        _aosoa_m.resize( 0 );
-        _aosoa_theta.resize( 0 );
+        _aosoa_m.resize( n_local );
+        _aosoa_theta.resize( n_local );
         _init_timer.stop();
     }
 
@@ -717,7 +717,7 @@ class Particles<MemorySpace, PMB, TemperatureDependent, Dimension>
     {
         // Forward arguments to standard or custom particle creation.
         base_type::createParticles( std::forward<Args>( args )... );
-        _aosoa_temp.resize( 0 );
+        _aosoa_temp.resize( n_local );
     }
 
     auto sliceTemperature()

--- a/src/CabanaPD_Particles.hpp
+++ b/src/CabanaPD_Particles.hpp
@@ -559,10 +559,11 @@ class Particles<MemorySpace, LPS, TemperatureIndependent, Dimension>
         _init_timer.stop();
     }
 
-    template <class ExecSpace>
-    void createParticles( const ExecSpace& exec_space )
+    template <typename... Args>
+    void createParticles( Args&&... args )
     {
-        base_type::createParticles( exec_space );
+        // Forward arguments to standard or custom particle creation.
+        base_type::createParticles( std::forward<Args>( args )... );
         _init_timer.start();
         _aosoa_m.resize( 0 );
         _aosoa_theta.resize( 0 );
@@ -711,10 +712,11 @@ class Particles<MemorySpace, PMB, TemperatureDependent, Dimension>
         init_temp();
     }
 
-    template <class ExecSpace>
-    void createParticles( const ExecSpace& exec_space )
+    template <typename... Args>
+    void createParticles( Args&&... args )
     {
-        base_type::createParticles( exec_space );
+        // Forward arguments to standard or custom particle creation.
+        base_type::createParticles( std::forward<Args>( args )... );
         _aosoa_temp.resize( 0 );
     }
 

